src/main/java/soot/jimple/internal/AbstractIntLongBinopExpr.java

```diff
@@ -62,6 +56,26 @@ public Type getType() {
     if (tyLong.equals(t1) && tyLong.equals(t2)) {
       return tyLong;
     }
+
+    // in dotnet enums are value types, such as myBool = 1 is allowed in CIL
+    if (Options.v().src_prec() == Options.src_prec_dotnet) {
+      if (t1 instanceof RefType) {
```

Comment:
Das ist in der ``AbstractFloatBinopExpr`` auch so drin. Bitte nach Duplikaten schauen und als Hilfsmethode in die gemeinsame Oberklasse übernehmen.